### PR TITLE
fix setMinZoom/setMaxZoom

### PR DESCRIPTION
**Pull request by cs09g**
#6746

### CODE REVIEW

File: src/ol/view.js [hunk 1]
```diff
@@ -177,6 +178,20 @@ ol.View.prototype.applyOptions_ = function(options) {
...
+    if (isNaN(nextResolution)) {
+      nextResolution = properties[ol.ViewProperty.RESOLUTION];
+      if (isNaN(nextResolution)) {
+        nextResolution = undefined;
+      }
+    }
+
+    if (this.resolutions_ && nextResolution) {
+      properties[ol.ViewProperty.RESOLUTION] = ol.math.clamp(
+        nextResolution,
```
Comment (cs09g) [inline src/ol/view.js]: build warning:
```
ERR! compile found : *
ERR! compile required : number
```

File: src/ol/view.js [hunk 2]
```diff
@@ -1055,9 +1071,8 @@ ol.View.createResolutionConstraint_ = function(options) {
       options.zoomFactor : defaultZoomFactor;
 
   if (options.resolutions !== undefined) {
-    var resolutions = options.resolutions;
-    maxResolution = resolutions[0];
-    minResolution = resolutions[resolutions.length - 1];
+    maxResolution = minZoom !== undefined ? resolutions[minZoom] : resolutions[0];
+    minResolution = maxZoom !== undefined ? resolutions[maxZoom] : resolutions[defaultMaxZoom];
     resolutionConstraint = ol.ResolutionConstraint.createSnapToResolutions(
         resolutions);
```
Comment (cs09g) [inline src/ol/view.js]: build warning:
```
ERR! compile found : <Array<number>|null|undefined>
ERR! compile required : <Array<number>|null>
```

**Comment by ahocevar** [comment]
Is this ready for review? Can you please squash your commits into one?

**Comment by cs09g** [comment]
Sure. I'll check the code again.

**Pull request closed without merging**